x-pack/filebeat/input/awss3/input.go

```diff
@@ -411,6 +411,10 @@ func getProviderFromDomain(endpoint string, ProviderOverride string) string {
 func pollSqsWaitingMetric(ctx context.Context, receiver *sqsReader) {
 	t := time.NewTicker(time.Minute)
 	defer t.Stop()
+	// Initialize the metric to 0 at the start of the one minute time interval to avoid
+	// giving misleading metric value -1 even though SQS messages are processed.
+	// The value will be updated every minute
+	receiver.metrics.sqsMessagesWaiting.Set(int64(0))
```

Comment:
I think this is more misleading to report 0. We now have documented that "-1 indicates the metric is uninitialized". And uninitialized is the state that it is in until the timer executes the first time. So reporting -1 is correct. Reporting 0 means it observed the SQS queue ApproximateMessageCount to be zero, but that's not what happened.

If you want this metric to be initialized earlier then I would refactor this to execute the first `GetQueueAttributes` request before starting the timer.